### PR TITLE
GCPerfSim Improvements

### PR DESCRIPTION
**Pull request by mrsharm**
Improvement of https://github.com/dotnet/performance/pull/3322

1. Removed some dead code and added explanation for why some code is uncommented.
2. Fixed compilation issue due to the nonexistence of the POH before .NET 5.
3. Replaced the StreamWriter with simple Console Writes.

### CODE REVIEW

File: src/benchmarks/gc/GCPerfSim/GCPerfSim.cs [hunk 1]
```diff
@@ -2536,14 +2519,6 @@ void MakeObjectAndMaybeSurvive()
...
         (ITypeWithPayload item, ObjectSpec spec) = MakeObjectAndTouchPage();
 
-        // if (isLarge && args.lohPauseMeasure)
-        // {
-        //     stopwatch.Stop();
-        //     lohAllocPauses.Add(stopwatch.Elapsed.TotalMilliseconds);
-        // }
-
-        // Thread.Sleep(1);
-
```
Comment (Maoni0) [inline src/benchmarks/gc/GCPerfSim/GCPerfSim.cs:2546]: this should be kept since you are keeping the commented out section above

**Review by Maoni0 — DISMISSED** [review]
LGTM with nit above

**Review by Maoni0 — APPROVED** [review]

**Comment by mrsharm** [comment]
@cincuranet, can we please merge this PR, as well? Thanks!

**Comment by cincuranet** [comment]
@mrsharm Yep.

**Comment by mrsharm** [comment]
@cincuranet - thanks for all your help!

**Pull request merged**